File: test-suite/src/test/java/io/micronaut/docs/inject/typed/V8Engine.java
package io.micronaut.docs.inject.typed;

import io.micronaut.context.annotation.Bean;

import jakarta.inject.Singleton;

// tag::class[]
@Singleton
@Bean(typed = Engine.class) // <1>
public class V8Engine implements Engine {  // <2>
    @Override
    public String start() {
        return "Starting V8";
    }

    @Override
    public int getCylinders() {
        return 8;
    }
}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/inject/typed/V8Engine.kt
package io.micronaut.docs.inject.typed

import io.micronaut.context.annotation.Bean
import jakarta.inject.Singleton

// tag::class[]
@Singleton
@Bean(typed = [Engine::class]) // <1>
class V8Engine : Engine { // <2>
    override fun start(): String {
        return "Starting V8"
    }

    override val cylinders: Int = 8
}
// end::class[]
